dev/run-black-linter.sh

```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+
+# Exclude proto files because they are auto-generated
+black --check --line-length=100 --exclude=mlflow/protos .
+exit_code="$?"
+if [[ "$exit_code" != "0" ]]; then
+  echo "Python lint failed. Run 'black --line-length=100 --exclude=mlflow/protos .' from your"\
+    "checkout of MLflow (note the trailing '.' at the end of the command, corresponding to the"\
+    "current directory) to autoformat Python code"
```

Comment:
Should we include this command in another script, like `./autoformat.sh` for ease of use? I assume this would need to be run essentially every iteration of the devloop, and it seems a bit of a pain to be told that your code is badly formatted, to run another command to fix it, but not be told what was actually wrong with it.

Reply:
Alternately, we could provide an `--autoformat` option to `lint.sh` itself

Reply:
+1 to --autoformat option to lint.sh!

Reply:
or we can use `pyproject.toml` and put black options in there so `black .`  becomes equivalent `black <options specified in pyproject.toml> .`

https://black.readthedocs.io/en/stable/pyproject_toml.html

Reply:
Both of these ideas sound good! Adding an option to `./lint.sh` seems preferable since users are already running it, as Apurva mentioned. Maybe an option like `--fix-black` since after running `./lint.sh --fix-black` other linters (e.g. pylint) can still fail with errors that need to be fixed manually

Reply:
Another advantage of `pyproject.toml` is that an editor like `vscode` can detect `pyproject.toml` and apply auto-formatting with options specified in it.